src/Mvc/Mvc.Core/src/ApiBehaviorOptions.cs

```diff
@@ -39,12 +41,23 @@ public class ApiBehaviorOptions : IEnumerable<ICompatibilitySwitch>
     /// When enabled, the following sources are inferred:
     /// Parameters that appear as route values, are assumed to be bound from the path (<see cref="BindingSource.Path"/>).
     /// Parameters of type <see cref="IFormFile"/> and <see cref="IFormFileCollection"/> are assumed to be bound from form.
+    /// Parameters that are complex (<see cref="ModelMetadata.IsComplexType"/>) and are registered in the DI Container (<see cref="IServiceCollection"/>) are assumed to be bound from the services <see cref="BindingSource.Services"/>, unless this
```

Comment:
Is the precedence order the same in minimal i.e is `IServiceProviderIsService` only used for complex types?